### PR TITLE
Add more unit test cases for getUpdatedTrackArtists

### PR DESCRIPTION
**Pull request by derat**
Test more edge cases, particularly when the track credits are visually identical to the release credits but reference different entities.

---

There are lots of weird edge cases not exercised by the existing test that I added in #3168. I'd like to add coverage for some of these before making any changes to `getUpdatedTrackArtists`.

### CODE REVIEW

**Comment by derat** [comment]
@reosarevok and @mwiencek, mind taking a look? I'm not sure what the "correct" behavior is in some of these cases, but I figured it'd be good to document how the code handles them right now.

**Review by reosarevok — APPROVED** [review]
Seems sensible to me, not sure if some should change as suggested, @mwiencek might have an opinion. The ones that are inconsistent between feat and no feat should change at least I guess :)

**Comment by derat** [comment]
@mwiencek, any thoughts about this, or does it look fine to you?

@brainzbot, retest this please

**Review by mwiencek — APPROVED** [review]
Insofar as this documents the current behavior, I think it's fine to merge, though I'm not sure I fully understand all of the real-world situations in which the two comments above describe the behavior you'd want every time. I guess that's why they are called edge cases. :)

**Comment by mwiencek** [comment]
@brainzbot retest this please

File: root/static/scripts/tests/edit/utility/getUpdatedTrackArtists.js [hunk 1]
```diff
@@ -122,6 +129,112 @@ test('getUpdatedTrackArtists', function (t) {
         name('C', '3'),
       ],
     },
+    {
+      desc: 'old name but diff. entity',
+      track: [name('A', '1')],
+      oldRel: [name('A', '2')],
+      newRel: [name('B', '3')],
+      want: [name('B', '3')],
```
[the remark this answers is not in the document]
Comment (derat) [reply root/static/scripts/tests/edit/utility/getUpdatedTrackArtists.js:137]: Yep, that's correct. I think I've seen this come about if there was an earlier cleanup that tried to fix an incorrectly-credited release (using A1 rather than A2, in this case) but missed some tracks.

File: root/static/scripts/tests/edit/utility/getUpdatedTrackArtists.js [hunk 2]
```diff
@@ -122,6 +129,112 @@ test('getUpdatedTrackArtists', function (t) {
...
+        name('B', '2', ' feat. '),
+        name('C', '3'),
+      ],
+    },
+    {
+      desc: 'new name but diff. entity',
+      track: [name('A', '1')],
+      oldRel: [name('B', '2')],
+      newRel: [name('A', '3')],
+      want: [name('A', '1')],
```
[the remark this answers is not in the document]
Comment (derat) [reply root/static/scripts/tests/edit/utility/getUpdatedTrackArtists.js:189]: Yes, also correct. This seems pretty subtle to me, and I'm not sure that the difference in behavior makes sense (I certainly don't find it intuitive as an editor).

I think that `getUpdatedTrackArtists` is in need of a rewrite; I just want to improve the test coverage so it's easier to see what changes. :-)
